__tests__/index.tests.js

```diff
@@ -39,3 +43,155 @@ describe('error 404', ()=>{
+          
+              return request(app)
+              .get('/apii')
+              .expect(404)
+     
+          })
+      })
+      describe('GET /api/articles/:article_id', () => {
+        test('GET:200 sends an object with the article corresponding to the id given to the client', () => {
+          
+          return request(app)
+            .get(`/api/articles/1`)
+            .expect(200)
+            .then((response) => {
+              
+              expect(response.body.article).toEqual(expect.any(Object));
+              expect(Object.keys(response.body.article[0])).toEqual(
+                expect.arrayContaining(["author",
+                "title",
+                "article_id",
+                "body",
+                "topic",
+                "created_at",
+                "votes",
+                "article_img_url"  ]))
+                expect(response.body.article[0].article_id).toBe(1)
+              
+                       
+            });
+          }
+        )
+        test('GET:404 on bad path', ()=>{
+          
+          return request(app)
+          .get('/api/articless/1')
+          .expect(404)
+ 
+      })
+      test('GET:404 sends an appropriate and error message when given a valid but non-existent id', () => {
+        return request(app)
+          .get('/api/articles/9999999')
+          .expect(404)
+          .then((response) => {
+            console.log(response.body.msg)
+            expect(response.body.msg).toBe('article doesn\'t exist');
+          });
+      });  
+      }
+      )
+      describe('GET /api/articles', () => {
+        test('GET:200 sends an object with the articles', () => {
+          
+          return request(app)
+            .get(`/api/articles`)
+            .expect(200)
+            .then((response) => {
+              return response.body.results.forEach(element=>{
+                
+                expect(Object.keys(element)).toEqual(
+                  expect.arrayContaining(["author",
+                  "title",
+                  "article_id",
+                  
+                  "topic",
+                  "created_at",
+                  "votes",
+                  "article_img_url",
+                "comment_count"]))
+              })
+                
+                       
+            });
+          }
+        )
+        test('GET:404 on bad path', ()=>{
+          
+          return request(app)
+          .get('/api/articless')
+          .expect(404)
+ 
+      })
+      
+      }
+      )
+      describe('GET /api/articles/:article_id/comments', () => {
+        test('GET:200 sends an object with the article corresponding to the id given to the client', () => {
+          
+          return request(app)
+            .get(`/api/articles/1/comments`)
+            .expect(200)
+            .then((response) => {
+              console.log(response.body.results)
+              expect(response.body.results).toEqual(expect.any(Object));
```

Comment:
I would take out the ```expect.any(Object)``` test here for the reasons August explained before, and keep the ```Object.keys``` part below